dmoj/executors/filesystem_policies.py

```diff
@@ -0,0 +1,114 @@
+import os
+from typing import List, Union
+
+
+class ACCESS_MODE:
+    NONE = 0
+    EXACT = 1
+    RECURSIVE = 2
+
+
+class Dir:
+    def __init__(self):
+        self.access_mode = ACCESS_MODE.NONE
+        self.map = {}
+
+
+class File:
+    pass
+
+
+class ExactFile:
+    def __init__(self, path: str):
+        self.path = path
+
+
+class ExactDir:
+    access_mode = ACCESS_MODE.EXACT
+
+    def __init__(self, path: str):
+        self.path = path
+
+
+class RecursiveDir:
+    access_mode = ACCESS_MODE.RECURSIVE
+
+    def __init__(self, path: str):
+        self.path = path
+
+
+Rule = Union[ExactFile, ExactDir, RecursiveDir]
+
+
+class FilesystemPolicy:
+    def __init__(self, rules: List[Rule]):
+        self.root = Dir()
+        for rule in rules:
+            self._add_rule(rule)
+
+    def _add_rule(self, rule: Rule):
```

Comment:
Everything that has no return type should specify `-> None` as the type, otherwise mypy will usually infer the type as `Any`.

Reply:
What functions/variables should have type annotations?

Reply:
And I guess, how much annotation is required/useful

Reply:
Everything that can be annotated should be annotated to the extent possible. https://github.com/DMOJ/judge-server/blob/master/dmoj/judge.py is a good example.